fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalIntersect.java

```diff
@@ -98,4 +104,23 @@ public LogicalIntersect withNewOutputs(List<NamedExpression> newOutputs) {
         return new LogicalIntersect(qualifier, newOutputs, regularChildrenOutputs,
                 Optional.empty(), Optional.empty(), children);
     }
+
+    @Override
+    public FunctionalDependencies computeFuncDeps(Supplier<List<Slot>> outputSupplier) {
+        FunctionalDependencies.Builder builder = new FunctionalDependencies
+                .Builder(child(0).getLogicalProperties().getFunctionalDependencies());
```

Comment:
for intersect, right side fd also can be propagated.

Reply:
Yeah, you are right